src/shared/util.js

```diff
@@ -695,6 +746,22 @@ class Util {
     return [xt, yt];
   }
 
+  static applyTransformInPlace(p, m) {
+    const [p0, p1] = p;
+    p[0] = p0 * m[0] + p1 * m[2] + m[4];
+    p[1] = p0 * m[1] + p1 * m[3] + m[5];
+  }
```

Comment:
*Possibly a stupid idea, but:*
To avoid having to add an almost duplicate method, how about changing the existing `applyTransform` to support the new use-case as well?

```diff
diff --git a/src/shared/util.js b/src/shared/util.js
index cfbd9995d..d445a7aca 100644
--- a/src/shared/util.js
+++ b/src/shared/util.js
@@ -689,10 +689,11 @@ class Util {
   }

   // For 2d affine transforms
-  static applyTransform(p, m) {
-    const xt = p[0] * m[0] + p[1] * m[2] + m[4];
-    const yt = p[0] * m[1] + p[1] * m[3] + m[5];
-    return [xt, yt];
+  static applyTransform(p, m, out = []) {
+    const [p0, p1] = p;
+    out[0] = p0 * m[0] + p1 * m[2] + m[4];
+    out[1] = p0 * m[1] + p1 * m[3] + m[5];
+    return out;
   }

   static applyInverseTransform(p, m) {
```

Reply:
I did that but then I had to create a variable (one for p and out).
Ideally, we should get rid of `applyTransform` and only use `applyTranformInPlace` (whatever the name is).
In some large pdfs, we create a way too much temp objects such like that which lead to waste some time in gc-ing them.
We should get rid of getCurrentTransform and probably track the current transform ourselves in order to avoid to have to create an object each time we call ctx.getTransform().
So my idea was to keep for now this "duplication" and just remove one of them once we're able to apply in place.
Anyway, I'm fine to apply your patch, just tell me what you prefer.

Reply:
Sure, we can keep the patch as-is.